squire_lib/src/accounts.rs

```diff
@@ -0,0 +1,29 @@
+use uuid::Uuid;
+use mtgjson::model::deck::Deck;
+use settings;
+
+enum SharingPermissions {
+    decklist{name: String, deck: Deck},
+    user_name(String),
+}
+
+struct UserId {
```

Comment:
This works and is how we had been doing typed ids, but I would recommend looking at how the identifiers are defined in `identifiers.rs`. There, we have a `TypeId` and implement several methods on it. That type is also generic over an "unused" type parameter. This allows us to implement all the things we generally want a typed id to do only once.

In that file, I would add `use` statements to pull in the account types and add the following 
```rust
pub type UserAccountId = TypeId<SquireAccount>;
pub type OrgAccountId = TypeId<OrganizationAccount>;
```